restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -515,11 +526,24 @@ public JsonNode getJsonAtPath(
       return null;
     }
     RawDocument rawDoc = docs.get(0);
+    List<Row> rows = rawDoc.rows();
+    boolean filtered = false;
+    if (!fields.isEmpty()) {
+      List<String> convertedPath =
+          path.stream()
+              .map(pathSeg -> DocumentServiceUtils.convertArrayPath(pathSeg.getPath()))
+              .collect(Collectors.toList());
+
+      rows = fieldsFilter.filter(rows, fields, convertedPath);
+      rows = rows.stream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
Don't get this, how can a Row be null here?

Reply:
Good catched. I missed that. The old fields filter method used to add nulls, but the refactored code does not. Will fix.

Reply:
removed null filtering